### ISSUE TITLE
Any possible support nested bucket? 

### ISSUE THREAD

**Issue by navono** [timshannon/bolthold#71]
Currently, use `Upsert` or `Insert ` to insert or update value is convenience without create bucket, but in some cases need nested bucket that bolthold not currently supported.

Is there any plane to support nested bucket?

**Comment by timshannon** [comment]
It might be possible.  Right now you have direct access to the underlying bolt DB, so you can use nested buckets, just not with the querying interface.

What is the scenario where this would be useful?  You can insert and query nested structs in Bolthold without needing nested buckets.

**Comment by navono** [comment]
Sorry for late response.

I have some nested data like this:

|- project1(bucket)
|    |__ device_type_1(bucket)
|    |        |__ device_1(kv)
|    |        |__ device_2(kv)
|    |
|    |__ device_type_2(bucket)
|    |        |__ device_1
|    |        |__ device_2
|    |
|- project2(bucket)
|

store same type data in same bucket, so can quickly locate the target bucket with some specific role(like name + id), also update or insert item in same bucket.

This is my scenario. Yes, with direct access to the underlying bolt DB, I can create nested nested buckets, but it will lose all of the benifit with bolthold, can't use exists nested bucket  and query with `bolthold`.

Any alternative way to achieve this?

**Comment by timshannon** [comment]
So do you actually need querying at the `project` or `device_type` levels?  If not, then you could simply put them in different database files.  It would also allow for better performance since you won't have to wait on writes.

**Comment by timshannon** [comment]
I think I've got this sorted out, just need to write some tests and I'll merge the changes.

**Comment by navono** [comment]
I have already know the project bucket name, and need query `device_type`. All about this is need a way to setup the custom bucket name as the operation bucket, then update or insert item on it.

Different database files is the alternative way, but adds administrative overhead, need a way to join queries from different databases.

**Comment by timshannon** [comment]
Nested Bucket support Added in #72

**Comment by timshannon** [comment]
pushed to master, open up a new issue if you run into anything.

Thanks,
